File: src/main/java/cal/fstg/java/controller/EmployerController.java
package cal.fstg.java.controller;
//
//import cal.fstg.java.dao.EmployerRepository;
//import cal.fstg.java.model.Employer;
//import cal.fstg.java.service.EmployerService;
//import org.springframework.beans.factory.annotation.Autowired;
//import org.springframework.web.bind.annotation.*;
//
//import java.util.List;
//import java.util.Optional;

//@RestController
//@RequestMapping("/employers")
//public class EmployerController {
//
//    private final EmployerService employerService;
//
//    @Autowired
//    public EmployerController(EmployerService employerService) {
//        this.employerService = employerService;
//    }
//
//    @GetMapping("/get/all")
//    public List<Employer> getAll() {
//        return employerService.getAll();
//    }
//
//    @GetMapping("/get/{id}")
//    public Optional<Employer> getOne(@PathVariable long id) {
//        return employerService.getOne(id);
//    }
//
//    @PostMapping("/save")
//    public void save(@RequestBody Employer employer) {
//        employerService.save(employer);
//    }
//
//    @DeleteMapping("/delete/{id}")
//    public void delete(@PathVariable long id) {
//        employerService.delete(id);
//    }
//}

File: src/main/kotlin/cal/fstg/stagerkotlin/controller/EmployerController.kt
package cal.fstg.stagerkotlin.controller

import cal.fstg.stagerkotlin.dao.EmployerRepository
import cal.fstg.stagerkotlin.model.Employer
import cal.fstg.stagerkotlin.service.EmployerService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/employers")
class EmployerController(@Autowired private val employerService: EmployerService) {

    @GetMapping("/get/all")
    fun getAll(): List<Employer> = employerService.getAll()

    @GetMapping("/get/{id}")
    fun getOne(@PathVariable id: Long) = employerService.getOne(id)

    @PostMapping("/save")
    fun save(@RequestBody employer: Employer) = employerService.save(employer)

    @DeleteMapping("/delete/{id}")
    fun delete(@PathVariable id: Long) = employerService.delete(id)

}